### ISSUE TITLE
dtest auth_test.TestAuth.kill_the_node_with_the_auth_info_test fails with std::runtime_error (Can't find a column family with UUID)

### ISSUE THREAD

**Issue by bhalevy** [scylladb/scylla#4115]
Seen in master dtest debug [#1004](http://jenkins.cloudius-systems.com:8080/view/master/job/scylla-master/job/dtest/1004/label=scylla-dtest,mode=debug,smp=2/testReport/auth_test/TestAuth/kill_the_node_with_the_auth_info_test/) 

```
Unexpected error in node1 node log: ["ERROR 2019-01-17 01:07:52,685 [shard 0] storage_proxy - Exception when communicating with 127.0.28.2: std::runtime_error (Can't find a column family with UUID 5bc52802-de25-35ed-aeab-188eecebb090)"]
```

**Comment by slivne** [comment]
This is in a debug build - lets clear the release issues first - it may be a timing issue related to debug.

**Comment by mamoit** [comment]
Seen this with docker image `scylladb/scylla:3.0.1`

**Comment by slivne** [comment]
@mamoit can you please provide the logs from your docker machine.

**Comment by mamoit** [comment]
This is a dev cluster, so I can try stuff if needed.
I had a happy 3 node cluster, when one of the machines was preempted (GCP).
When the node came back up with the same persistent volume it started spamming warningss and errors of which:
```
WARN  2019-01-30 15:23:06,835 [shard 2] storage_proxy - Failed to apply mutation from 10.32.1.5#2: std::_Nested_exception<no_such_column_family> (Can't find a column family with UUID ba570630-2480-11e9-b330-000000000000): std::out_of_range (_Map_base::at)
```
and
```
ERROR 2019-01-30 15:18:55,716 [shard 1] storage_proxy - Exception when communicating with 10.32.1.5: std::runtime_error (Can't find a column family with UUID b9412960-2480-11e9-a7a4-000000000003)
```
Are the only warnings or errors in the recovering node.
partial log: [scylla-1.log](https://github.com/scylladb/scylla/files/2813221/scylla-1.log)
let me know if you need anything more.

**Comment by tgrabiec** [comment]
Looks like https://github.com/scylladb/scylla/issues/420. Does restarting all nodes fix the problem?

In order to confirm this I need full logs from both nodes (10.32.1.5 and this one), which include the point in time when tables were first created.

**Comment by tgrabiec** [comment]
If this is really #420, then https://github.com/scylladb/scylla/commit/40a30d4129465edf4efc3bf3e88ade3ff6ead8ed (master only) should cause the cluster to auto-heal without the need for a restart.

**Comment by mamoit** [comment]
I will be able to test restarting the failing node tomorrow.
But I have a faint idea that i tried it and it went back to the same loop.

**Comment by mamoit** [comment]
Ok, I was able to reproduce the issue.
1. I started 3 scylla nodes (0, 1 and 2), they all came up, no problem [scylla-0.txt](https://github.com/scylladb/scylla/files/2821483/scylla-0.txt) [scylla-1.txt](https://github.com/scylladb/scylla/files/2821487/scylla-1.txt) [scylla-2.txt](https://github.com/scylladb/scylla/files/2821482/scylla-2.txt) (the logs for 0 and 2 include all the trial)
2. Started 3 kairos nodes, all with the same config, and they all tried to init the keyspace), this may cause a race condition and this issue?
3. Scylla node 1 started spamming the `Failed to apply mutation` message
4. I restarted node 1 and it kept on spamming `Failed to apply mutation` [scylla-1-restarted.txt](https://github.com/scylladb/scylla/files/2821473/scylla-1-restarted.txt), check logs for node 0 and 2 to see the other nodes reaction.

It didn't happen quite identically, I'm going to tear it down and try with one kairos node to avoid a race condition and try to get the scylla node restart issue.
